staging/src/k8s.io/apiserver/pkg/server/filters/wrap.go

```diff
@@ -29,11 +29,19 @@ import (
 func WithPanicRecovery(handler http.Handler) http.Handler {
 	return withPanicRecovery(handler, func(w http.ResponseWriter, req *http.Request, err interface{}) {
 		if err == http.ErrAbortHandler {
-			// honor the http.ErrAbortHandler sentinel panic value:
-			//   ErrAbortHandler is a sentinel panic value to abort a handler.
-			//   While any panic from ServeHTTP aborts the response to the client,
-			//   panicking with ErrAbortHandler also suppresses logging of a stack trace to the server's error log.
-			return
+			klog.Warningf("Aborting the request since ErrAbortHandler was received (%v) on %v %v. This might result in a EOF error on the client side", err, req.Method, req.RequestURI)
+			// Honor the http.ErrAbortHandler sentinel panic value:
+			//
+			// If ServeHTTP panics, the server (the caller of ServeHTTP) assumes
+			// that the effect of the panic was isolated to the active request.
+			// It recovers the panic, logs a stack trace to the server error log,
+			// and either closes the network connection or sends an HTTP/2
+			// RST_STREAM, depending on the HTTP protocol. To abort a handler so
+			// the client sees an interrupted response but the server doesn't log
+			// an error, panic with the value ErrAbortHandler.
+			//
+			//
+			panic(http.ErrAbortHandler)
```

Comment:
panicking with `http.ErrAbortHandler` (well any panic would do) will actually close the underlying connection.

Previously `errConnKilled` was captured, logged and the connection was left intact since calling `WriteHeader` multiple times is not supported by the std library.

I think that before closing the connection (before panicking) we could actually try to send a message to the client.

Reply:
> I think that before closing the connection (before panicking) we could actually try to send a message to the client.

Even if we did, the msg would be ignored by the client (golang) library `On error, any Response can be ignored`
https://golang.org/src/net/http/client.go?s=20422:20474#L575

Reply:
Where did we catch errConnKilled ?